sonoff/xsns_48_chirp.ino

```diff
@@ -0,0 +1,484 @@
+    I2cWriteReg(addr, CHIRP_SLEEP);
+}
+
+/********************************************************************************************/
+
+// void ChirpSleepAll(void) {
+//     for (uint8_t i = 0; i < chirp_found_sensors; i++) {
+//       if (chirp_sensor[i].version) { 
+//         ChirpSleep(chirp_sensor[i].address);
+//         }
+//     }
+// }
+
+// /********************************************************************************************/
+
+// void ChirpAutoWakeAll(void) {
+//     for (uint8_t i = 0; i < chirp_found_sensors; i++) {
+//       if (chirp_sensor[i].version && !chirp_sensor[i].explicitSleep) { 
+//         ChirpReadVersion(chirp_sensor[i].address);
+//         }
+//     }
+// }
+
+/********************************************************************************************/
+
+void ChirpSelect(uint8_t sensor) {
+  if(sensor < chirp_found_sensors) { //TODO: show some infos
+    chirp_current = sensor;
+    AddLog_P2(LOG_LEVEL_DEBUG, "CHIRP: Sensor %u now active.", chirp_current);
+  }
+  if (sensor == 255) {
+    AddLog_P2(LOG_LEVEL_DEBUG, "CHIRP: Sensor %u active at address 0x%x.", chirp_current, chirp_sensor[chirp_current].address);
+  }
+}
+
+/********************************************************************************************/
+
+bool ChirpMeasureLight(void) {
+  for (uint8_t i = 0; i < chirp_found_sensors; i++) {
+    if (chirp_sensor[i].version && !chirp_sensor[i].explicitSleep) { 
+      uint8_t lightReady = I2cRead8(chirp_sensor[i].address, CHIRP_GET_BUSY);
+      AddLog_P2(LOG_LEVEL_DEBUG, "CHIRP: busy status for light for sensor %u", lightReady);
+      if (lightReady == 1) {
+        return false; // a measurement is still in progress, we stop everything and come back in the next loop = 1 second
+      }
+      AddLog_P2(LOG_LEVEL_DEBUG, "CHIRP: init measure light for sensor %u", i);
+      I2cWriteReg(chirp_sensor[i].address, CHIRP_MEASURE_LIGHT); 
+      }
+  }
+  return true; // we could read all values (maybe at different times, but that does not really matter) and consider this job finished
+}
+
+/********************************************************************************************/
+
+void ChirpReadCapTemp() { // no timeout needed for both measurements, so we do it at once
+    for (uint8_t i = 0; i < chirp_found_sensors; i++) {
+      if (chirp_sensor[i].version && !chirp_sensor[i].explicitSleep) { 
+        AddLog_P2(LOG_LEVEL_DEBUG, "CHIRP: now really read CapTemp for sensor at address 0x%x", chirp_sensor[i].address);
+        chirp_sensor[i].moisture = I2cRead16(chirp_sensor[i].address, CHIRP_GET_CAPACITANCE);
+        chirp_sensor[i].temperature = I2cRead16(chirp_sensor[i].address, CHIRP_GET_TEMPERATURE); 
+        }
+    }
+}
+
+/********************************************************************************************/
+
+bool ChirpReadLight() {   // sophisticated calculations could be done here
+  bool success = false;
+  for (uint8_t i = 0; i < chirp_found_sensors; i++) {
```

Comment:
You can use `uint32_t` in loops, in sometimes make the code smaller, see https://github.com/arendst/Sonoff-Tasmota/wiki/Sensor-API#user-content-keeping-esp8266-code-compact